addons/mail/static/src/new/discuss/channel_selector.js

```diff
@@ -66,18 +66,18 @@ export class ChannelSelector extends Component {
                     this.state.selectedPartners,
                 ]);
                 const suggestions = results.map((data) => {
-                    this.partnerService.insert(data);
+                    this.personaService.insert({ ...data, type: "partner" });
                     return {
                         classList: "o-mail-channel-selector-suggestion",
                         label: data.name,
                         partner: data,
                     };
                 });
-                if (this.messaging.currentPartner.name.includes(cleanedTerm)) {
+                if (this.store.self.name.includes(cleanedTerm)) {
```

Comment:
not related to current PR, but why do we have to handle this in JS rather than have im_search return it?